ShellPkg/Library/UefiShellAcpiViewCommandLib/Parsers/Ras2/Ras2Parser.c

```diff
@@ -114,5 +115,23 @@ ParseAcpiRas2 (
       sizeof (EFI_ACPI_RAS2_PCC_DESCRIPTOR)
       );
     Offset += sizeof (EFI_ACPI_RAS2_PCC_DESCRIPTOR);
+    Count++;
   } // while
+
+  // Check counts match and print error if not
+  if (Count > *Ras2PccDescriptors) {

```

Comment:
NIT: Maybe a shorter version:

```
BOOLEAN IsTooMany;

IsTooMany = Count > *Ras2PccDescriptors;
Print (
  L"\nERROR: Descriptors provided are too %s (%d vs %d)",
  IsTooMany ? "many" : "few",
  Count,
  *Ras2PccDescriptors
  );
```

Reply:
I could indeed do it that way - less in the way of strings compiled into acpivew.